src/main/java/org/kairosdb/core/datastore/KairosDatastore.java

```diff
@@ -528,6 +544,8 @@ public int getSampleSize()
 				m_results.add(aggregatedGroup);
 			}
 
+			span.finish();
```

Comment:
Won't we have broken spans if the code above throws?

Reply:
+1
Just make sure you try-with-resources on the Scope, not the Span.  The Span still needs will need a finally { span.finish() } or else you won't be able to use it in the catches.